credoai/modules/model_modules/privacy.py

```diff
@@ -76,127 +124,195 @@ def run(self):
             Key: metric name
             Value: metric value
         """
+        attacks_to_run = SUPPORTED_MEMBERSHIP_ATTACKS
+        if self.attack_feature:
+            attacks_to_run = attacks_to_run | SUPPORTED_ATTRIBUTE_ATTACKS
 
         attack_scores = {}
-        for attack_name, attack_info in self.SUPPORTED_PRIVACY_ATTACKS.items():
+        for attack_name, attack_info in attacks_to_run.items():
             attack_scores[attack_name] = self._general_attack_method(attack_info)
 
         # Best model = worst case
-        membership_inference_worst_case = max(attack_scores.values())
+        attack_scores["membership_inference_attack_score"] = max(
+            [v for k, v in attack_scores.items() if "Membership" in k]
+        )
 
-        attack_scores[
-            "membership_inference_attack_score"
-        ] = membership_inference_worst_case
+        if self.attack_feature:
+            attack_scores["attribute_inference_attack_score"] = max(
+                [v for k, v in attack_scores.items() if "Attribute" in k]
+            )
 
         self.results = attack_scores
 
         return self
 
-    def prepare_results(self):
-        """Prepares results for export to Credo AI's Governance App
-
-        Structures a subset of results for export as a dataframe with appropriate structure
-        for exporting. See credoai.modules.credo_module.
-
-        Returns
-        -------
-        pd.DataFrame
-
-        Raises
-        ------
-        NotRunError
-            If results have not been run, raise
-        """
-        if self.results is not None:
-            return Series(self.results, name="value")
-        else:
-            raise NotRunError("Results not created yet. Call 'run' to create results")
-
     def _general_attack_method(self, attack_details):
         """
         General wrapper for privacy modules from ART.
 
         There are 2 types of modules: the ones leveraging machine learning and
-        the rule based ones. The former require an extra fit step, and a further
-        split of tranining and test so that there is no leakage during the assessment
-        phase.
+        the rule based ones. The former require an extra fit step.
 
         Parameters
         ----------
         attack_details : dict
-            Map of all the supported ART privacy modules and their relative type.
+            Dictionary containing all the attack details
```

Comment:
It's still not clear what `attack details` are. What are their components and how are they used? Rather than defining that, I think writing out the arguments here would be easier.

Reply:
It seems you want to keep them as "attack details" so you can pass them around. If so, that's fine, but you may want to create a namedtuple or dataclass